src/cmds/scripts/pbs_habitat.in

```diff
@@ -655,10 +673,10 @@ if [ "${PBS_START_SERVER:-0}" != 0 ] ; then
 			exit $ret
 		else
 			#Remove old dataservice directory
-			if [ `echo "${old_inst_dir}"| grep "pgsql.old"` ]; then
-				err=`rm -rf "${old_inst_dir}"`
+			if [ -d "${old_inst_dir}" ]; then
+				backupdir "$old_inst_dir" "$PBS_HOME"
```

Comment:
we are calling backupdir() only in one place, why write a separate routine for that?
Also, the old_inst_dir is in $PBS_HOME. We need work out all the logic that we are in backupdir() routine.

Reply:
@prakashcv13 thank you for the review.
I see your point, here I am using function to use only at one place, but this function may require in the future and helps to make code self documented, what you?

Reply:
@riyazhakki - hmmm, but your sub routine is not generic - the "echo" statement is specific to an upgrade/downgrade. If you want to keep the subroutine - let us make the statement generic or remove it from the subroutine.

Reply:
@prakashcv13 I have addressed your comment in the latest commit, please have a look.